eclipse-che/che-plugin-registry
# chore: add pull-request check to ensure plugin sidecar has memory and CPU limitations

Signed-off-by: Vitaliy Gulyy <vgulyy@redhat.com>



### What does this PR do?
Introduces a PR-check to prevent adding the plugins without or with wrong CPU and memory requests/limitations.


### What issues does this PR fix or reference?

https://github.com/eclipse/che/issues/16685

### How to test this PR?

- clone this project, switch to `resource-limit-pr-check` branch
- go to `tools/automation`
- build with `yarn prepare`
- check plugins by `yarn node ./lib/check-resource-limits.js`
- change / remove cpu/memory limits/requests in `che-theia-plugins.yaml` and check plugins again

### PR Checklist

[As the author of this Pull Request I made sure that:](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#pull-request-template-and-its-checklist)


### Reviewers

Reviewers, please comment how you tested the PR when approving it.
